### PR TITLE
Update vdirsyncer (+ one dep), khard

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -2772,12 +2772,12 @@ in {
   };
 
   click-log = buildPythonPackage rec {
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: While you're at it, please move this expression to a separate file in `python-modules/`

**Comment by teto** [comment]
Didn't see this PR and I had prepared one myself. Could you pick this hunk please https://github.com/teto/nixpkgs/commit/fcbbf45a6451ee27fa69deb857983a246179c510#diff-fa94f0db0527b004ae3809fa214083a0R27 ? I am afraid the locale might leak in a normal environement when it's only needed for the tests.

**Comment by matthiasbeyer** [comment]
@teto I guess your hunk is outdated.

**Comment by matthiasbeyer** [comment]
@FRidh updated for latest versions.

**Comment by teto** [comment]
I meant changing the LC_ALL variable to the test part where it's used:
 https://github.com/teto/nixpkgs/commit/fcbbf45a6451ee27fa69deb857983a246179c510#diff-fa94f0db0527b004ae3809fa214083a0L26 . I am worried about side effects (though there may be none ?! like does the LC_ALL get exported in nix-shell ?)

**Comment by teto** [comment]
Looking forward to this one
The failures on darwin are locale related. Not sure how it works there, the locale seems missing
```
nix/store/ab2iwb8ks5day43sw0580hdf54nfa4pv-bash-4.4-p12/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.utf8): No such file or directory

============================= test session starts ==============================

platform darwin -- Python 3.6.2, pytest-3.2.1, py-1.4.34, pluggy-0.4.0

rootdir: /private/tmp/nix-build-vdirsyncer-0.16.3.drv-0/vdirsyncer-0.16.3, inifile: setup.cfg

plugins: hypothesis-3.11.1, localserver-0.3.7, subtesthack-0.1.1

collected 486 items

tests/storage/test_filesystem.py ..............................................sss......ss.......

tests/storage/test_http.py .....

tests/storage/test_http_with_singlefile.py ......................................sssssss....sssss.ss

tests/storage/test_memory.py ......................................sssssss....sss...ss

tests/storage/test_singlefile.py ....................................................ss.ss

tests/storage/dav/test_caldav.py ssssssssssssssssssssssssssssssssssssssssssssssssssssss

tests/storage/dav/test_carddav.py ssssssssssssssssssssssssss

tests/storage/dav/test_main.py .................................

tests/system/cli/test_config.py .........

tests/system/cli/test_discover.py FFFFF

tests/system/cli/test_fetchparams.py F

tests/system/cli/test_repair.py FFFFF

tests/system/cli/test_sync.py FFFFFFFFsFFFFFFFFFFF

tests/system/cli/test_utils.py ..

tests/system/utils/test_main.py .....

tests/unit/test_exceptions.py .

tests/unit/test_metasync.py .........

tests/unit/test_repair.py ..........

tests/unit/cli/test_config.py .

tests/unit/cli/test_discover.py ......

tests/unit/cli/test_fetchparams.py .........

tests/unit/sync/test_status.py .

tests/unit/sync/test_sync.py ...............................

tests/unit/utils/test_vobject.py ..................

=================================== FAILURES ===================================

____________________________ test_discover_command _____________________________

tests/system/cli/test_discover.py:38: in test_discover_command

    assert not result.exception

E   assert not FileNotFoundError(2, "No such file or directory: 'locale'")

E    +  where FileNotFoundError(2, "No such file or directory: 'locale'") = <Result FileNotFoundError(2, "No such file or directory: 'locale'")>.exception

___________________ test_discover_different_collection_names ___________________

tests/system/cli/test_discover.py:91: in test_discover_different_collection_names

    assert not result.exception

E   assert not FileNotFoundError(2, "No such file or directory: 'locale'")

E    +  where FileNotFoundError(2, "No such file or directory: 'locale'") = <Result FileNotFoundError(2, "No such file or directory: 'locale'")>.exception

__________________________ test_discover_direct_path ___________________________

tests/system/cli/test_discover.py:134: in test_discover_direct_path

    assert not result.exception

E   assert not FileNotFoundError(2, "No such file or directory: 'locale'")

E    +  where FileNotFoundError(2, "No such file or directory: 'locale'") = <Result FileNotFoundError(2, "No such file or directory: 'locale'")>.exception

__________________ test_null_collection_with_named_collection __________________

tests/system/cli/test_discover.py:161: in test_null_collection_with_named_collection

    assert not result.exception

E   assert not FileNotFoundError(2, "No such file or directory: 'locale'")

E    +  where FileNotFoundError(2, "No such file or directory: 'locale'") = <Result FileNotFoundError(2, "No such file or directory: 'locale'")>.exception
```

File: pkgs/development/python-modules/click-log/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{ stdenv, buildPythonPackage, fetchurl, click }:
+
+buildPythonPackage rec {
+  version = "0.2.0";
+  name = "click-log-${version}";
+
+  src = fetchurl {
```
Comment (FRidh) [inline pkgs/development/python-modules/click-log/default.nix]: fetchPypi

**Comment by matthiasbeyer** [comment]
Outdated.

**Comment by teto** [comment]
Are you going to push an updated one ? was waiting for this one to solve a google calendar + vdirsyncer problem

**Comment by matthiasbeyer** [comment]
@teto was already updated in https://github.com/NixOS/nixpkgs/pull/31041

**Pull request closed without merging**